### ISSUE TITLE
Add more tests for report formatting

### ISSUE THREAD

**Issue by gavv** [gavv/httpexpect#165]
Currently we have only one such test: TestE2EReportNames. It checks that test and request names are included into formatted report.

It would be useful to add more similar tests and check that some other elements from AssertionContext and AssertionFailure are included into the report: Path, Errors, Actual / Expected / Reference / Delta values (of a few different types, including AssertionRange and AssertionList).

This tests should not be strict to the report format, ideally they should not break often when we're changing formatting. They should not make many assumptions about the report structure, but instead should check whether it contains desired pieces of text. This is how TestE2EReportNames works currently.

**Comment by makejjay** [comment]
Hi @gavv, I'd like to work on this.

**Comment by gavv** [comment]
You're welcome, thanks!

**Comment by gavv** [comment]
@makejjay Hi, do you still have plans on this?

**Comment by makejjay** [comment]
@gavv Yes, I expect to have some tests ready for review this week. Happy New Year!

**Comment by gavv** [comment]
Cool, thank you. No hurry, just checking.  Happy new year :)

**Comment by gavv** [comment]
Unassigning so that someone could pick this up.

**Comment by L-block** [comment]
I can take this one

**Comment by gavv** [comment]
Thanks!
